tools/fw-update/rs-fw-update.cpp

```diff
@@ -342,6 +377,19 @@ int main(int argc, char** argv) try
         std::cout << std::endl << "Updating device: " << std::endl;
         print_device_info(d);
 
+        // If device is D457 connected by MIPI connector
+        if( strcmp( d.get_info( RS2_CAMERA_INFO_PRODUCT_ID ), "ABCD" ) == 0 &&
+            usb_type.compare( "unknown" ) == 0)
+        {
+            if( unsigned_arg.isSet() )
+            {
+                std::cout << std::endl << "Only signed FW is currently supported for MIPI devices" << std::endl;
+                return EXIT_FAILURE;
+            }
+
+            return write_to_mipi_device( fw_image );
```

Comment:
Can't we do it under the hood? not in the tool itself?
Does the viewer support FW update as like as D400 USB?
At what state does the FW update finish?
If we query the fw version will we see the new one? 

Can you share a video of the update process?

Thanks!

Reply:
As discussed, too complicated for the required time slot, we will do in in the tool for now.